### DISCUSSION TITLE
请求一些2D图片、特效的示例代码

### DISCUSSION THREAD

**Discussion by shar0** [ejoy/ant#37] [Q&A] [answered]
想了解Ant引擎是否能用于开发2D游戏。

**Comment by cloudwu** [comment, marked as answer]
目前并没有设计 2d 的渲染管线，虽然做一个应该不难。只是开发团队暂时精力有限，我们需要一个实际的 2d 游戏项目驱动，才可能去实现和完善这样一个东西。

但是如果只是玩玩的话，可以用游戏 UI 完成。引擎集成了 RmlUI ，而 RmlUI 本身就可以用来做 2d 游戏。官方就有游戏的示例。

引擎可以运行 test/rmlui/main.lua 来查看一个简单的游戏 UI demo 。

我估计开发起来的感觉会有点像 web 游戏？通过修改 web 控件的节点的 css 属性来完成移动、旋转、变换。

**Comment by cloudwu** [comment]
https://mikke89.github.io/RmlUiDoc/pages/tutorials/window_template.html  这是 RmlUI 官方带的小游戏教程。

不过 Ant 是自己维护的分支，已经很不一样了，教程的做法无法照搬。

**Reply by shar0** [reply]
有无推荐的学习路径、或是从何处看起的参考，来自行尝试扩展简单的2D Sprite基础渲染功能？

**Reply by cloudwu** [reply]
我已经尽最大努力在 wiki 上写文档。自定义渲染管线的文档优先级相当低。

**Reply by shar0** [reply]
大佬加油👍

**Comment by vkensou** [comment]
相机不能设置成正交投影吗？

**Reply by cloudwu** [reply]
做 2d 游戏的话，不需要把相机设置成正交投影。因为那本质上还是 3d 逻辑。2d 绘图的话，应该单独写一个着色器。

https://github.com/ejoy/ant/wiki/RenderPipeline#%E8%87%AA%E5%AE%9A%E4%B9%89%E6%9D%90%E8%B4%A8

这里介绍了 2d 矩形图块的着色器，仓库里也有。

**Reply by vkensou** [reply]
啊，我只是简单提议一下。如果楼主想尝试下2D游戏，可以用正交相机+无光shader，这样即时没有专门的2D管线，也可以做2D游戏。
